src/protocol/parsers/detector.cpp

```diff
@@ -63,47 +65,85 @@ void ProtocolDetector::scan() {
         // Not found on UDP, now try all available serial ports
         auto ports = QSerialPortInfo::availablePorts();
 
-        for (auto it = ports.begin(); it != ports.end(); ++it) { // Scan all available ports
+        // Scan all available ports
+        for (auto& port : ports) {
+
+            // Check if port can be opened
+            if(!canOpenPort(port, 500)) {
+                qCDebug(PING_PROTOCOL_PROTOCOLDETECTOR) << "Couldn't open port" << port.portName();
+                continue;
+            }
+
+            QSerialPort p(port);
 
-            QSerialPort p(*it);
             for (const int baudrate : {115200, 921600}) {
-                qCDebug(PING_PROTOCOL_PROTOCOLDETECTOR) << "Probing Serial" << it->portName() << baudrate;
-                if (p.open(QIODevice::ReadWrite)) { // Attempt to open port
-                    p.setBaudRate(baudrate);
-
-                    // Probe
-                    p.write(reinterpret_cast<const char*>(req.msgData.data()), (uint16_t)req.msgData.size());
-                    p.waitForBytesWritten();
-
-                    bool detected = false;
-                    int attempts = 0;
-
-                    while (!detected && attempts < 10) { // Try to get a valid response, timeout after 40 ms
-                        p.waitForReadyRead(50);
-                        auto buf = p.readAll();
-                        for (auto byte = buf.begin(); byte != buf.end(); ++byte) {
-                            detected = _parser.parseByte(*byte) == PingParser::NEW_MESSAGE;
-                            if (detected) {
-                                break;
-                            }
+                qCDebug(PING_PROTOCOL_PROTOCOLDETECTOR) << "Probing Serial" << port.portName() << baudrate;
+
+                p.open(QIODevice::ReadWrite);
```

Comment:
shouldn't we _still_ check for success on this call to open?

Reply:
Yes, this is done in the next line, I'll this two.